Fix Bottom Overflow issue (#9213)
flutter/website: src/codelabs/implicit-animations.md

```

{{site.alert.secondary}}

>   Only set the start and end values of `opacity`.

Reviewer: I think the previous wording here was important, since it's not telling them to only set start and end, but explaining why they don't have to worry about the in-between.

```

{{site.alert.secondary}}
  You only need to set the start and end values of `opacity`.
  The `AnimatedOpacity` widget manages everything in between.
{{site.alert.end}}

### Fade-in (complete)

Here's the example with the completed changes you've made.
Run this example then click **Show details** to trigger the animation.

{% include docs/implicit-animations/fade-in-complete.md %}

### Putting it all together

The [Fade-in text effect][] example demonstrates the following features
of the `AnimatedOpacity` widget.

- It listens for state changes to its `opacity` property.
- When the `opacity` property changes,
  it animates the transition to the new value for `opacity`.
- It requires a `duration` parameter to define how long
  the transition between the values should take.

{{site.alert.note}}

> 

Reviewer: ```suggestion
```

  - Implicit animations can only animate the
    properties of a parent stateful widget.
    The preceding example enables this with the
    `FadeInDemo` widget that extends `StatefulWidget`.

  - The `AnimatedOpacity` widget only animates the `opacity` property.
    Some implicitly animated widgets can animate many properties

>     at the same time. The following example illustrates this.

Reviewer: ```suggestion
    at the same time. The following example showcases this.
```

**The example begins with no animation code**.
It starts with a [Material App][] home screen that contains:

- A `Container` widget configured with a
 `borderRadius`, `margin`, and `color`.
  These properties are setup to be regenerated 
  each time you run the example.
- A **Change** button that does nothing when clicked.

### Shape-shifting (starter code)

To start the example, click **Run**.

{% include docs/implicit-animations/shape-shifting-starter-code.md %}

### Animate color, borderRadius, and margin with AnimatedContainer

This section contains a list of steps you can use to add an
implicit animation to the [shape-shifting starter code][].
After completing each step, you can also run the
[complete shape-shifting example][] with the changes already made.

The [shape-shifting starter code][] assigns
each property in the `Container` widget a random value.
Associated functions generate the relevant values:

- The `randomColor()` function generates a
  `Color` for the `color` property
- The `randomBorderRadius()` function generates a
  `double` for the `borderRadius` property.
- The `randomMargin()` function generates a
  `double` for the `margin` property.

> This section uses the `AnimatedContainer` widget to:

Reviewer: At this point, I'm lost on what "this section" is referring. Perhaps "The following steps"?

```suggestion
The following steps use the `AnimatedContainer` widget to:
```

### Putting it all together

The [Fade-in text effect][] example demonstrates the following features

> of the `AnimatedOpacity` widget.

Author: I don't think we always need to use the colon to introduce something. The phrasing already covers it.

The [Fade-in text effect][] example demonstrates the following features
of `AnimatedOpacity`:

- `AnimatedOpacity` listens for state changes in its `opacity` property.
- Whenever `opacity` changes, `AnimatedOpacity` automatically animates the
  widget's transition to the new value for `opacity`.
- `AnimatedOpacity` requires a `duration` parameter to define the time it takes
  to animate the transition between an old `opacity` value and a new one.

{{site.alert.secondary}}
  Implicit animations can only animate properties of a parent

>   `StatefulWidget`. This example begins with the `FadeInDemo` widget that

Reviewer: This change loses an important connection between the two parts. The first part serves to explain the second part, but that's not clear now.

  extends `StatefulWidget`.

>   The `AnimatedOpacity` animates a single property: `opacity`.

Reviewer: Without the "Notice also" this reads a bit weird, as I'd expect it to be related to the other text in the same aside, but it's a separate point.

Author: **reply:** The whole thing is in a "note", so "Notice..." is redundant. I have a fix.

### Animate color, borderRadius, and margin with AnimatedContainer

This section contains a list of steps you can use to add an
implicit animation to the [shape-shifting starter code][].
After the steps, you can also run the
[shape-shifting complete][] example with the changes already made.

In the [shape-shifting starter code][],
each property in the `Container` widget (`color`,
`borderRadius`, and `margin`)
is assigned a value by an associated function (`randomColor()`,
`randomBorderRadius()`, and `randomMargin()` respectively).
To refactor this code, use an `AnimatedContainer` widget to:

- Generate new values for `color`, `borderRadius`,
  and `margin` whenever the user clicks **Change**.

> - Animate the transition to the new values for `color`,

Reviewer: It's not that they use `AnimatedContainer` to generate new values, but rather it allows them to transition to the new values.

#### 2. Set starting values for animated properties

> The `AnimatedContainer` widget transitions between old and new values of
> its properties when they change.

Reviewer: ```suggestion
The `AnimatedContainer` widget transitions between
old and new values of its properties when they change.
```

To contain the behavior triggered when the user clicks **Change**,
create a `change()` method.

> The `change()` method can use the `setState()` method to set new values

Reviewer: Nice.

### Shape-shifting (complete)

Here’s the example with the completed changes you’ve made.
Run the code and click **Change** to trigger the animation.
Each time you click **Change**, the shape animates to its new values
for `margin`, `borderRadius`, and `color`.

{% include docs/implicit-animations/shape-shifting-complete.md %}

### Using animation curves

The preceding examples show how:

> - Implicit animations allow you to transition between values for
>   specific widget properties.

Reviewer: ```suggestion
- Implicit animations allow you to animate the transition between
  values for specific widget properties.
```

- The `duration` parameter allows you to set how long the animation
  takes to complete.

Implicit animations also allow you to control changes to **the rate**
of an animation that occurs during the set `duration`.
To define this change in rate,
set the value of the `curve` parameter to
a [`Curve`][], such as one declared in the [`Curves`][] class.

The preceding examples did not specify a value for the `curve` parameter.
Without a specified curve value,
the implicit animations apply a [linear animation curve][].

> Add a `curve` parameter to the [complete shape-shifting example][].

Reviewer: ```suggestion
Specify a value for the `curve` parameter in
the [complete shape-shifting example][].
```

> Watch how the animation changes when you pass the

Author: **nitpick:** Small update.
```suggestion
The animation changes when you pass the
```

the [shape-shifting complete][] example builds upon [fade-in complete][].

- Some implicit animations, like the `AnimatedOpacity` widget,
  only animate one property.
  Others, like the `AnimatedContainer` widget, can animate many properties.
- Implicit animations transition between the old and new property values
  using the provided `curve` and `duration`.
- If you do not specify a `curve`,
  implicit animations default to a [linear curve][].

## What's next?

Congratulations, you've finished the codelab!
To learn more, check out these suggestions:

- Try the [animations tutorial][].
- Learn about [hero animations][] and [staggered animations][].
- Checkout the [animation library][].
- Try another [codelab][].

[AnimatedContainer]: {{site.api}}/flutter/widgets/AnimatedContainer-class.html
[AnimatedOpacity]: {{site.api}}/flutter/widgets/AnimatedOpacity-class.html
[animation library]: {{site.api}}/flutter/animation/animation-library.html
[animations tutorial]: {{site.url}}/ui/animations/tutorial
[codelab]: {{site.url}}/codelabs
[Curves]: {{site.api}}/flutter/animation/Curve-class.html
[duration]: {{site.api}}/flutter/widgets/ImplicitlyAnimatedWidget/duration.html

> [`easeInOutBack``]: {{site.api}}/flutter/animation/Curves/easeInOutBack-constant.html

Reviewer: ```suggestion
[`easeInOutBack`]: {{site.api}}/flutter/animation/Curves/easeInOutBack-constant.html
```

             ElevatedButton(
```

When you pass the `Curves.easeInOutBack` constant to the `curve` property
of the `AnimatedContainer` widget, watch how
the rates of change for `margin`, `borderRadius`, and `color`

> follow the curve that constant defined.

Reviewer: Right before the snippet, you already said "watch how". I think you can drop the "watch how" and switch to calling attention to what actually changed, similar to "notice" in the original text.

<div id="animation_1_play_button_"></div>
<video id="animation_1" style="width:464px; height:192px;" loop="">
  <source src="{{site.flutter-assets}}/animation/curve_ease_in_out_back.mp4" type="video/mp4">
</video>

### Putting it all together

The [complete shape-shifting example][] animates transitions between
values for `margin`, `borderRadius`, and `color` properties.
The `AnimatedContainer` widget animates changes to any of its properties.
These include those you didn't use such as `padding`, `transform`,
and even `child` and `alignment`!
By showing additional capabilities of implicit animations,
the [complete shape-shifting example][] builds upon
[fade-in complete][] example.

To summarize implicit animations:

- Some implicit animations, like the `AnimatedOpacity` widget,
  only animate one property.
  Others, like the `AnimatedContainer` widget, can animate many properties.

> - Implicit animations transition between the old and new value of a property
>   when it changes using the provided `curve` and `duration`.

Reviewer: ```suggestion
- Implicit animations animate the transition between the
  old and new value of a property when it
  changes using the provided `curve` and `duration`.
```

The [shape-shifting complete][] example animates transitions between values for
`margin`, `borderRadius`, and `color` properties.

> The `AnimatedContainer` animates changes to any of its properties.

Reviewer: `AnimatedContainer` is a type of object, not an object, so:

```suggestion
`AnimatedContainer` animates changes to any of its properties.
```

These include those you didn't use such as `padding`, `transform`,
and even `child` and `alignment`!
By showing additional capabilities of implicit animations,
the [shape-shifting complete][] example builds upon [fade-in complete][]:

> - Some implicit animations, like `AnimatedOpacity` only animate one property,
>   while others like `AnimatedContainer` can animate many properties.

Reviewer: ```suggestion
- Some implicit animations, like `AnimatedOpacity`, only animate one property,
  while others, like `AnimatedContainer`, can animate many properties.
```
